code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,601 @@
+		return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DisIntegrate(var/atom/movable/target)
+	new /obj/effect/swarmer/disintegration(get_turf(target))
+	do_attack_animation(target)
+	changeNext_move(CLICK_CD_MELEE)
+	target.ex_act(3)
+	return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DisperseTarget(var/mob/living/target)
+	if(target != src)
+		src << "<span class='info'>Attempting to remove this being from our presence.</span>"
+		if(src.z != ZLEVEL_STATION)
+			src << "<span class='warning'>Our bluespace transceiver cannot locate a viable bluespace link, our teleportation abilities are useless in this area.</span>"
+			return
+		if(do_mob(src, target, 30))
+			var/cycle
+			for(cycle=0,cycle<100,cycle++)
+				var/random_location = locate(rand(37,202),rand(75,192),ZLEVEL_STATION)//Drunk dial a turf in the general ballpark of the station
+				if(istype(random_location, /turf/simulated/floor))
+					var/turf/simulated/floor/F = random_location
+					if(F.air)
+						var/datum/gas_mixture/A = F.air
+						if(A.oxygen >= 16 && !A.toxins && A.carbon_dioxide < 10 && !A.trace_gases.len)//Can most things breathe in this location?
+							if((A.temperature > 270) && (A.temperature < 360))//Not too hot, not too cold
+								var/pressure = A.return_pressure()
+								if((pressure > 20) && (pressure < 550))//Account for crushing pressure or vaccuums
+									if(ishuman(target))//If we're getting rid of a human, slap some zipties on them to keep them away from us a little longer
+										var/obj/item/weapon/restraints/handcuffs/cable/zipties/Z = new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+										var/mob/living/carbon/human/H = target
+										Z.apply_cuffs(H, src)
+									do_teleport(target, F, 0)
+									playsound(src,'sound/effects/sparks4.ogg',50,1)
+									break
+			return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DismantleMachine(var/obj/machinery/target)
+	do_attack_animation(target)
+	src << "<span class='info'>We begin to dismantle this machine. We will need to be uninterrupted.</span>"
+	var/obj/effect/swarmer/dismantle/D = new /obj/effect/swarmer/dismantle(get_turf(target))
+	D.pixel_x = target.pixel_x
+	D.pixel_y = target.pixel_y
+	D.pixel_z = target.pixel_z
+	if(do_mob(src, target, 100))
+		src << "<span class='info'>Dismantling complete.</span>"
+		var/obj/item/stack/sheet/metal/M = new /obj/item/stack/sheet/metal(target.loc)
+		M.amount = 5
+		if(target.component_parts && target.component_parts.len)
+			for(var/obj/item/I in target.component_parts)
+				I.forceMove(M.loc)
+		var/obj/effect/swarmer/disintegration/N = new /obj/effect/swarmer/disintegration(get_turf(target))
+		N.pixel_x = target.pixel_x
+		N.pixel_y = target.pixel_y
+		N.pixel_z = target.pixel_z
+		target.dropContents()
+		src << "[target]"
+		if(istype(target, /obj/machinery/computer))
+			var/obj/machinery/computer/C = target
+			src << "[C][C.circuit]"
+			if(C.circuit)
+				C.circuit.loc = M.loc
```

Comment:
@Markolie yeah i tried to use force move earlier but it runtime also..I did not realize the whole textpath bit...